### PR TITLE
[Paper] Add function to calculate metrics from posteriors

### PR DESCRIPTION
**Pull request by SUKI-O**
Add function to calculate metrics from posteriors from saved runs in order to avoid re-running the entire code for different metrics. 

Changes proposed in this pull request:

-

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

File: sktree/tests/test_utils.py
Comment (adam2392) [inline sktree/tests/test_utils.py:1]: Here I would test with 

- `FeatureImportanceClassifier` with the HonestForest setup we have (i.e. MIGHT, or CoMIGHT)
- Something else like SVM from sklearn

Note that posteriors returned from something like sklearn has a diff shape.

Comment (adam2392) [reply sktree/tests/test_utils.py:1]: You can test against metric functions that take in "y_score":

- http://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html
- https://scikit-learn.org/stable/modules/generated/sklearn.metrics.top_k_accuracy_score.html#sklearn.metrics.top_k_accuracy_score
- our internal implementation of mutual information

A positive test is just a "smoke test" to make sure it runs and the value is "expected". Then a negative test could be where you change something that you can foresee being a common error-surface in our workflow and make sure we error-check properly.
